File: Data Structures & Algorithms/append-characters-to-string-to-make-subsequence/submission-0.cpp
class Solution {
public:
    int appendCharacters(string s, string t) {
        
        int n=s.size();
        int m=t.size();
        
        int i=0,j=0;
        while(i<n&&j<m){
           
           if(s[i]==t[j]){
            i++;
            j++;
           }
           else{              
              i++;
           }
           cout<<i<<" "<<j<<"\n";
        }

        if(j==m) return 0;
        cout<<j<<" "<<m;   
        return m-j;


    }
};

File: Data Structures & Algorithms/first-unique-character-in-a-string/submission-0.java
class Solution {
    public int firstUniqChar(String s) {
        int []alp=new int[26];

        Arrays.fill(alp,-1);

        for(int i=0;i<s.length();i++){
          
          if(alp[s.charAt(i)-'a']==-1){
            alp[s.charAt(i)-'a']=i;
          }
          else {
            alp[s.charAt(i)-'a']=s.length();
          }

          
        }
        int mini=Integer.MAX_VALUE;
        for(int i=0;i<26;i++){
            if(alp[i]!=-1&&alp[i]!=s.length()){

            mini=Math.min(mini,alp[i]);
            }

        }
        return mini==Integer.MAX_VALUE?-1:mini;
        
    }
}